srcts/src/bindings/input/text.ts

```diff
@@ -50,20 +50,32 @@ class TextInputBindingBase extends InputBinding {
   }
 
   subscribe(el: TextHTMLElement, callback: (x: boolean) => void): void {
-    $(el).on(
-      "keyup.textInputBinding input.textInputBinding",
-      // event: Event
-      function () {
-        callback(true);
-      }
-    );
-    $(el).on(
+    const $el = $(el);
+    const updateOn = $el.data("update-on") || "input";
+
+    if (updateOn === "input") {
+      $el.on(
+        "keyup.textInputBinding input.textInputBinding",
+        // event: Event
+        function () {
+          callback(true);
+        }
+      );
+    }
+
+    $el.on(
       "change.textInputBinding",
       // event: Event
       function () {
         callback(false);
       }
     );
```

Comment:
For `NumberInputBinding`, this `change` event can get triggered:

- programmatically, via `updateNumericInput()` on the server
- by clicking on the up/down arrows, or pressing up/down keys on the keyboard.

When we enable the option, we'll want the callback to be invoked when receiving the server message, but not when the up/down arrow event causes it.

Reply:
I handled this in 0f0e4e593d01ab5eb7c7c529867dd3020f09bd3a by adding extra data to the change event emitted by the `.receiveMessage()` method, whose presence signals that the origin of the event was a server-side message. Then, when `updateOn === "blur"`, we ignore change events that aren't server driven.

Reply:
One thing that's worth considering is the possibility that someone would trigger a `change` event on a programmatically via JS, like if they use shinyjs. Requiring `fromServer: true` could make that stop working.

Reply:
Good point; I'll think about this some more